homeassistant/components/deconz/switch.py

```diff
@@ -17,7 +17,7 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
     gateway.entities[DOMAIN] = set()
 
     @callback
-    def async_add_switch(lights):
+    def async_add_switch(lights=gateway.api.lights.values()):
```

Comment:
Not sure if that is a good or dangerous move. A good move in case you can call `async_add_switch()` 10x times without having a bad side affect, bad move if you should call this function only once without arguments and useless if never call this function with other as that value.

Reply:
It should never be an issue since the first thing done is to make sure to register the unique id for the platform https://github.com/home-assistant/core/blob/8caa177ba19141710708a00947e524f450667223/homeassistant/components/deconz/deconz_device.py#L58

Reply:
How often is `async_add_switch` called without arguments? Is it only the instance below?

Reply:
Once during startup and then when calling the device_refresh service async_add_switch will also be called without arguments

Reply:
Ok to go?